internal/services/storage/storage_account_resource.go

```diff
@@ -469,6 +469,7 @@ func resourceStorageAccount() *pluginsdk.Resource {
 						"logging": {
 							Type:     pluginsdk.TypeList,
 							Optional: true,
+							Computed: true,
```

Comment:
we should add a todo to remove this in 4.0, since we can remove `enabled` within the nested block and determine it based on the presence/omission of the block - as the items within it can only be configured when enabled?

Reply:
If the users have omit `minute_metrics` block, the API still returns it back. There is no way in the `Read` to conditionally set the `minute_metrics` based on the config, as [`Read` only has access to state](https://github.com/hashicorp/terraform-plugin-go/blob/main/tfprotov5/internal/tfplugin5/tfplugin5.proto#L233). So it appears that the `Computed: true` is still needed here.

In regard to the `enabled` for `minute_metrics` and `hour_metrics`, yes we can remove it in 4.0. But it doesn't apply to the `logging`, as there is not a single toggle in it, it has 3 toggles for read, write, delete.